central/complianceoperator/v2/checkresults/service/service_impl.go

```diff
@@ -157,6 +158,22 @@ func (s *serviceImpl) GetComplianceScanResultsCount(ctx context.Context, query *
 	}, nil
 }
 
+// GetComplianceOverallClusterCount returns scan results count
+func (s *serviceImpl) GetComplianceOverallClusterCount(ctx context.Context, query *v2.RawQuery) (*v2.CountComplianceScanResults, error) {
+	parsedQuery, err := search.ParseQuery(query.GetQuery(), search.MatchAllIfEmpty())
+	if err != nil {
+		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to parse query %v", err)
+	}
+
+	count, err := s.complianceResultsDS.ComplianceClusterStatsCount(ctx, parsedQuery)
+	if err != nil {
+		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to retrieve compliance scan results count for query %v", query)
```

Comment:
[nit] InvalidArgs isn't always accurate (for example if you got two results for a count query).